### PR TITLE
Add support for request and response customizers

### PR DESCRIPTION
**Pull request by geoand**
The idea is to use these to replicate some of the PME functionality
such as dependencyOverride and dependencyExclusion

### CODE REVIEW

**Comment by geoand** [comment]
test this please

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentService.java [hunk 1]
```diff
@@ -30,4 +30,46 @@ GAV getProject() {
 
 		String getAlignedVersionOfGav(GAV gav);
 	}
+
+	interface RequestCustomizer {
+
+		Request customize(Request request);
+
+		// Integer.MIN_VALUE is the max order
```
Comment (metacosm) [inline analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentService.java]: Could you clarify the comment to specify how the order works with respect to priority of customization, please?

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentServiceFactory.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package org.jboss.gm.analyzer.alignment;
+
+import java.util.Collections;
+
+public final class AlignmentServiceFactory {
```
Comment (metacosm) [inline analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentServiceFactory.java]: Why is the class `final`?

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentService.java (hunk 1 above)
Comment (geoand) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentService.java]: Sure thing

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentServiceFactory.java (hunk 2 above)
Comment (geoand) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentServiceFactory.java]: https://stackoverflow.com/questions/32375149/is-it-mandatory-utility-class-should-be-final-and-private-constructor

Basically utility classes should be `final` (along with having a private constructor) in order to make it clear how they are supposed to be used. It's good practice, but I'll agree that it's actual usefulness is limited.

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentService.java (hunk 1 above)
Comment (geoand) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentService.java]: Done

**Comment by geoand** [comment]
retest this please

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentServiceFactory.java (hunk 2 above)
Comment (metacosm) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentServiceFactory.java]: OK, I never bothered but that makes sense. Learning new things every day! :)

**Comment by geoand** [comment]
Feel free to do so :P

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentServiceFactory.java (hunk 2 above)
Comment (geoand) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentServiceFactory.java]: We all are :)

**Pull request merged**